File: Dessert.java
/**
 * This class represents a Dessert, which is a MenuItem.
 *
 * @author Matthew Hino and Reece Teramoto
 * @version April 9, 2014
 **/
public class Dessert extends MenuItem
{
    // Instance variable
    private int calorieCount;

    /**
     * Default constructor for Dessert
     */
    public Dessert()
    {
        super();
        calorieCount = 0;
    }

    /**
     * Constructor for Dessert
     * 
     * @param   name, description, price, calorie count
     */
    public Dessert( String name, String description, double price, int initCalCount )
    {
        super(name, description, price);
        this.calorieCount = initCalCount;
    }

    /**
     * This method returns the calorieCount
     * 
     * @return   calorie count
     **/
    public int getCalCount()
    {
        return calorieCount;
    }

    /**
     * toString for Dessert
     * 
     * @return   string representation of Dessert (name, price, calorie count, description)
     **/
    public String toString()
    {
        String itemDescription = this.getName() + "\t\t\t$" + this.getPrice() + " Calories: " + this.getCalCount() 
            + "\n" + "Description: " + this.getDescription();
        return itemDescription;
    }

    /**
     * equals for Dessert
     * returns true if another MenuItem has the same name
     * @param   other menu item
     */
    public boolean equals(MenuItem otherMenuItem)
    {
        boolean equals;
        if( this.getName().equalsIgnoreCase(otherMenuItem.getName()) )
        {
            equals = true;
        }
        else
        {
            equals = false;
        }
        return equals;
    }
}

File: Dessert.m
classdef Dessert < MenuItem
    % DESSERT Class for dessert menu items
    % Extends MenuItem with calorie information
    %
    % Author: MATLAB version adapted from Matthew Hino and Reece Teramoto's Java code
    % Version: 2025
    
    properties (Access = private)
        calories
    end
    
    methods
        function obj = Dessert(name, description, price, calories)
            % Constructor for Dessert
            obj@MenuItem(name, description, price);
            if nargin > 3
                obj.calories = calories;
            else
                obj.calories = 300; % Default calories
            end
        end
        
        function calories = getCalories(obj)
            calories = obj.calories;
        end
        
        function setCalories(obj, calories)
            obj.calories = calories;
        end
        
        function str = toString(obj)
            str = sprintf('%-30s $%.2f Calories: %d\nDescription: %s', ...
                obj.name, obj.price, obj.calories, obj.description);
        end
    end
end